x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/JoinType.java

```diff
@@ -44,8 +44,8 @@ public static JoinType readFrom(StreamInput in) throws IOException {
             case 0 -> INNER;
             case 1 -> LEFT;
             case 2 -> RIGHT;
-            case 4 -> FULL;
-            case 5 -> CROSS;
+            case 3 -> FULL;
+            case 4 -> CROSS;
```

Comment:
Good catch. 
But should this not be a bug fix? 8.15 will have the wrong IDs (there should be one more BC for it, so in time).

Reply:
The test really caught it. In production we only use LEFT at the moment. Have for a while. It's less exciting than it looks.